File: app/src/main/java/org/breezyweather/weather/WeatherServiceSet.java
package org.breezyweather.weather;

import androidx.annotation.NonNull;

import javax.inject.Inject;

import org.breezyweather.common.basic.models.options.provider.WeatherSource;
import org.breezyweather.weather.services.*;

public class WeatherServiceSet {

    private final WeatherService[] mWeatherServices;

    @Inject
    public WeatherServiceSet(OpenMeteoWeatherService openMeteoWeatherService,
                             AccuWeatherService accuWeatherService,
                             MetNoWeatherService metNoWeatherService,
                             OpenWeatherWeatherService openWeatherWeatherService,
                             MfWeatherService mfWeatherService,
                             ChinaWeatherService chinaWeatherService) {
        mWeatherServices = new WeatherService[] {
                openMeteoWeatherService,
                accuWeatherService,
                metNoWeatherService,
                openWeatherWeatherService,
                mfWeatherService,
                chinaWeatherService
        };
    }

    @NonNull
    public WeatherService get(WeatherSource source) {
        switch (source) {
            case OPEN_METEO:
                return mWeatherServices[0];

            case ACCU:
                return mWeatherServices[1];

            case METNO:
                return mWeatherServices[2];

            case OPEN_WEATHER:
                return mWeatherServices[3];

            case MF:
                return mWeatherServices[4];

            case CHINA:
                return mWeatherServices[5];

            default:
                return mWeatherServices[1]; // ACCU
        }
    }

    @NonNull
    public WeatherService[] getAll() {
        return mWeatherServices;
    }
}

File: app/src/main/java/org/breezyweather/weather/WeatherServiceSet.kt
package org.breezyweather.weather

import org.breezyweather.common.basic.models.options.provider.WeatherSource
import org.breezyweather.weather.accu.AccuWeatherService
import org.breezyweather.weather.china.ChinaWeatherService
import org.breezyweather.weather.metno.MetNoWeatherService
import org.breezyweather.weather.mf.MfWeatherService
import org.breezyweather.weather.openmeteo.OpenMeteoWeatherService
import org.breezyweather.weather.openweather.OpenWeatherWeatherService
import javax.inject.Inject

class WeatherServiceSet @Inject constructor(
    openMeteoWeatherService: OpenMeteoWeatherService,
    accuWeatherService: AccuWeatherService,
    metNoWeatherService: MetNoWeatherService,
    openWeatherWeatherService: OpenWeatherWeatherService,
    mfWeatherService: MfWeatherService,
    chinaWeatherService: ChinaWeatherService
) {
    val all: Array<WeatherService> = arrayOf(
        openMeteoWeatherService,
        accuWeatherService,
        metNoWeatherService,
        openWeatherWeatherService,
        mfWeatherService,
        chinaWeatherService
    )

    operator fun get(source: WeatherSource) = when (source) {
        WeatherSource.OPEN_METEO -> all[0]
        WeatherSource.ACCU -> all[1]
        WeatherSource.METNO -> all[2]
        WeatherSource.OPEN_WEATHER -> all[3]
        WeatherSource.MF -> all[4]
        WeatherSource.CHINA -> all[5]
    }
}
